# On macOS, environ is not defined. So fixed it by using _NSGetEnviron()
# which is the proper way to do it.

diff --git a/src/spawn-posix.c b/src/spawn-posix.c
index 7de02a9..78a21a4 100644
--- a/src/spawn-posix.c
+++ b/src/spawn-posix.c
@@ -56,7 +56,11 @@
 #endif /*__linux__ */
 
 #include "gpgrt-int.h"
-
+#ifdef __APPLE__
+# include <crt_externs.h>
+#else
+extern char **environ;
+#endif
 
 /* Definition for the gpgrt_spawn_actions_t.  Note that there is a
  * different one for Windows.  */
@@ -342,7 +346,11 @@ my_exec (const char *pgmname, const char *argv[], gpgrt_spawn_actions_t act)
   _gpgrt_close_all_fds (3, act->except_fds);
 
   if (act->environ)
+#ifdef __APPLE__
+    *_NSGetEnviron() = act->environ;
+#else
     environ = act->environ;
+#endif
 
   if (act->atfork)
     act->atfork (act->atfork_arg);
